azure_functions_worker/dispatcher.py

```diff
@@ -294,14 +291,56 @@ async def _handle__worker_init_request(self, req):
                 result=protos.StatusResult(
                     status=protos.StatusResult.Success)))
 
-    async def _handle__worker_status_request(self, req):
+    async def _handle__worker_status_request(self, request):
         # Logging is not necessary in this request since the response is used
         # for host to judge scale decisions of out-of-proc languages.
         # Having log here will reduce the responsiveness of the worker.
         return protos.StreamingMessage(
-            request_id=req.request_id,
+            request_id=request.request_id,
             worker_status_response=protos.WorkerStatusResponse())
 
+    async def _handle__functions_metadata_request(self, request):
+        metadata_request = request.functions_metadata_request
```

Comment:
add error handling in case any error happens during this func call so grpc call wont be stuck and host can log the failed response to customer